src/core/packet.lua

```diff
@@ -99,6 +99,12 @@ function data (p) return p.data end
 -- Return packet data length.
 function length (p) return p.length end
 
+-- Set packet data length.
+function resize (p, len)
+   assert(len <= max_payload, "packet payload overflow")
+   p.length = len
+end
+
```

Comment:
I see two options:

1. Divide `resize` up into `grow` and `shrink` (makes the code slightly awkward and API kind of encourages branching, although not in this specific case) :

```
--- Set packet data length.
-function resize (p, len)
-   assert(len <= max_payload, "packet payload overflow")
-   p.length = len
+-- Grow packet data length.
+function grow (p, o)
+   local len = p.length
+   local new_len = len + o
+   assert(new_len <= max_payload, "packet payload overflow")
+   ffi.fill(p.data+len, o)
+   p.length = new_len
+end
+
+-- Shrink packet data length.
+function shrink (p, o)
+   local len = p.length
+   local new_len = len-o
+   assert(new_len >= 0, "packet payload underflow")
+   p.length = new_len
```

2. Always zero packet memory in `free` (gives a pretty solid protection against information leaks, performance impact?):

```
 function free (p)
    counter.add(engine.frees)
    counter.add(engine.freebytes, p.length)
+   -- Zero packet data
+   ffi.fill(p.data, p.length)
 [...]
```